### ISSUE TITLE
No 'version' key in the manifest file for custom integration 'meross_cloud

### ISSUE THREAD

**Issue by userMak** [albertogeniola/meross-homeassistant#218]
**Describe the bug**
No 'version' key in the manifest file for custom integration 'meross_cloud

**Your environment**
Operating System
Ubuntu 18.04.5 LTS
Docker version
20.10.5
Version
supervisor-2021.03.6
Newest Version
supervisor-2021.03.6
Channel
stable

**Logs taken when the issue happened**
```
Logger: homeassistant.loader
Source: loader.py:802
First occurred: 9:19:21 AM (1 occurrences)
Last logged: 9:19:21 AM

No 'version' key in the manifest file for custom integration 'meross_cloud'. This will not be allowed in a future version of Home Assistant. Please report this to the maintainer of 'meross_cloud' ---
```

**Comment by Raul-7-7** [comment]
Same issue, this will not be supported as of June, 2021
`WARNING (MainThread) [homeassistant.loader] No 'version' key in the manifest file for custom integration 'meross_cloud'. As of Home Assistant 2021.6, this integration will no longer be loaded. Please report this to the maintainer of 'meross_cloud'`

**Comment by Raul-7-7** [comment]
In case anyone runs into this issue later on, to fix this
modify your manifest.json (/custom_components/meross_cloud/manifest.json) and add the last line I have here.

{
  "domain": "meross_cloud",
  "name": "Meross Cloud IoT",
  "documentation": "https://www.home-assistant.io/components/meross_cloud",
  "dependencies": ["persistent_notification"],
  "codeowners": ["@albertogeniola"],
  "requirements": ["meross-iot==0.4.1.0rc7"],
  "config_flow": true,
  "quality_scale": "platinum",
  "version": "1.0"
}

**Comment by THATDONFC** [comment]
> In case anyone runs into this issue later on, to fix this
> modify your manifest.json (/custom_components/meross_cloud/manifest.json) and add the last line I have here.
> 
> {
> "domain": "meross_cloud",
> "name": "Meross Cloud IoT",
> "documentation": "https://www.home-assistant.io/components/meross_cloud",
> "dependencies": ["persistent_notification"],
> "codeowners": ["@albertogeniola"],
> "requirements": ["meross-iot==0.4.1.0rc7"],
> "config_flow": true,
> "quality_scale": "platinum",
> "version": "1.0"
> }

Don't forget the comma after the "quality_scale" line.

`"quality_scale": "platinum",`

**Comment by unlucio** [comment]
I had the same problem, added the version key and value.
The error got silenced, but I still cannot find the integration in the list, even after a number of restarts.
Any suggestion?

**Comment by Raul-7-7** [comment]
@unlucio did it disappear?
- Delete from Integration
- Delete from HACS
re-install see if that works.

**Comment by unlucio** [comment]
yeah, did that.
That's how I ended up with he missing version in the manifest 😂: removing it and reinstalling it with hacs :P

**Comment by Raul-7-7** [comment]
@unlucio removing and re-installing from HACS should NOT have given you version manifest. hmm interesting, I am not sure then, sorry bud.

**Comment by unlucio** [comment]
@Raul-7-7 yeah it's confusing.
I saw the error in the logs, come here to open a PR to add the version number, and found out the repo's manifest.json is actually correct.
So I manually added it to my installation and rebooted HA, but when I went to add the integration it's still nowhere to be found in the list.
Logs are clean tho (besides the usual warning for unsupported modules)

**Comment by Raul-7-7** [comment]
@unlucio can you post your manifest here? it might be that the formatting was lost.

**Comment by unlucio** [comment]
@Raul-7-7 I posted it in the bug report I opened not noticing this one:
https://github.com/albertogeniola/meross-homeassistant/issues/218

Wait, it's json, not yaml, it shouldn't care about formatting 🤔
and yes, the number of comas is correct, I wrote quite a bit of json in my life 😆
besides: if there was a missing coma it would have caused a parse error, and logs are clean :(

**Comment by swythan** [comment]
@Raul-7-7 I have the same issue as @unlucio

My (hand edited) manifest.json is currently:
``` json
{
  "domain": "meross_cloud",
  "name": "Meross Cloud IoT",
  "documentation": "https://www.home-assistant.io/components/meross_cloud",
  "dependencies": ["persistent_notification"],
  "codeowners": ["@albertogeniola"],
  "requirements": ["meross-iot==0.4.1.1rc1"],
  "config_flow": true,
  "quality_scale": "platinum",
  "version": "1.0.5"
}
``` 

Just before I got into trying to fix this (when it was still unable to load) I disabled the integration and it immediately disappeared (even with the "show disabled" option). I then removed it in HACS, and got into a cycle of changing version, uninstalling, manual editing, rebooting, etc, but I've not managed to get it back yet.

**Comment by Raul-7-7** [comment]
@unlucio nothing wrong with double checking! regardless of how many json you have written :) trying to help bud! 

@swythan that looks good, honestly I am not sure anymore, sorry :(  do you have a snapshot before you upgraded to latest HA? maybe restore that if you do? and go from there. 
I added the version to my manifest back in March (when I originally made my post) and just upgraded to latest HA without issue. Sorry I could not help more :(

**Comment by swythan** [comment]
> @swythan that looks good, honestly I am not sure anymore, sorry :( do you have a snapshot before you upgraded to latest HA? maybe restore that if you do? and go from there.

I've gone back to a snapshot from before the HA upgrade and meross is working again. TBH I only did the upgrade because I happened to see that it was available. I didn't realise I was on the bleeding edge! I think I'll leave it a bit before trying again.

**Comment by swythan** [comment]
I saw the 1.0.7 release and decided to give it a try. 

I installed the 1.0.7 version of meross_cloud and _then_ updated HA and it worked perfectly first time.  I can't provide evidence for doing it the other way around.

**Comment by albertogeniola** [comment]
Solved in from 1.0.7
